azure-servicebus/src/main/java/com/microsoft/azure/servicebus/primitives/MiscRequestResponseOperationHandler.java

```diff
@@ -234,4 +231,51 @@ else if(statusCode == ClientConstants.REQUEST_RESPONSE_NOCONTENT_STATUS_CODE ||
 			});
 		});		
 	}
+
+	public CompletableFuture<List<RuleDescription>> getRulesAsync(int skip, int top)
+	{
+		TRACE_LOGGER.debug("Fetching rules for entity '{}'", this.entityPath);
+		return this.createRequestResponseLink().thenComposeAsync((v) -> {
+			HashMap requestBodyMap = new HashMap();
+			requestBodyMap.put(ClientConstants.REQUEST_RESPONSE_SKIP, skip);
+			requestBodyMap.put(ClientConstants.REQUEST_RESPONSE_TOP, top);
+
+			Message requestMessage = RequestResponseUtils.createRequestMessageFromPropertyBag(
+					ClientConstants.REQUEST_RESPONSE_GET_RULES_OPERATION,
+					requestBodyMap,
+					Util.adjustServerTimeout(this.underlyingFactory.getOperationTimeout()));
+			CompletableFuture<Message> responseFuture =
+					this.requestResponseLink.requestAysnc(requestMessage, this.underlyingFactory.getOperationTimeout());
+			return responseFuture.thenComposeAsync((responseMessage) -> {
+				CompletableFuture<List<RuleDescription>> returningFuture = new CompletableFuture<>();
+
+				List<RuleDescription> rules = new ArrayList<RuleDescription>();
+				int statusCode = RequestResponseUtils.getResponseStatusCode(responseMessage);
+				if(statusCode == ClientConstants.REQUEST_RESPONSE_OK_STATUS_CODE)
+				{
+					Map responseBodyMap = RequestResponseUtils.getResponseBody(responseMessage);
+					Map[] rulesMap = (Map[])responseBodyMap.get(ClientConstants.REQUEST_RESPONSE_RULES);
+					for (Map ruleMap : rulesMap)
+					{
+						rules.add(RequestResponseUtils.decodeRuleDescriptionMap(ruleMap));
+					}
+
+					TRACE_LOGGER.debug("Fetched rules from entity '{}'", this.entityPath);
```

Comment:
>TRACE_LOGGER.debug("Fetched rules from entity '{}'", this.entityPath); [](start = 5, length = 70)

log the number of rules fetched in the statement. like fetched '{}' rule from the entity '{}', rules.size()..